parliament_challenge.py

```diff
@@ -0,0 +1,119 @@
+import requests
+import waitress
+import flask
+from flask import Flask, request, json
+
+
+def filter_speeches_dict(speeches):
+    """
+    Filtering speeches data for relevant keys.
+    Relevant keys are anforande_id, dok_datum,
+    parti, avsnittsrubrik, links and intressent_id.
+    """
+
+    speeches_list = []
+    for s in speeches:
+        speechesDict = {
+            "anforande_id": s.get("anforande_id"),
+            "dok_datum": s.get("dok_datum"),
+            "parti": s.get("parti"), 
+            "avsnittsrubrik": s.get("avsnittsrubrik"),
+            "links": [{ 
+                "rel": "speech",
+                "href": s.get("protokoll_url_www")
+                    }],
+            "intressent_id": s.get("intressent_id")
+            }
+        speeches_list.append(speechesDict)
+    return speeches_list
+
+
+def filter_member_dict(member):
+    """
+    Filtering member data for relevant keys.
+    Relevant keys are tilltalsnamn, valkrets,
+    bild_url_192 and intressent_id.
+    """
+
+    membersDict = {
+        "tilltalsnamn": member.get("tilltalsnamn"),
+        "valkrets": member.get("valkrets"),
+        "bild_url_192": member.get("bild_url_192"),
+        "intressent_id": member.get("intressent_id")
+    }
+    return membersDict
+
+
+def get_member_data(intressent_id):
+    """
+    Get a member data from the member's api
+    and filter for relevant keys.
+    """
+
+    domain = 'http://data.riksdagen.se'
+    members = 'personlista'
+    format_type = 'json'
+    response = requests.get('{}/{}/?iid={}&utformat={}'.format(domain, members, intressent_id, format_type))
+    if response.status_code == 200:
+        data = response.json()
+        member = data["personlista"]["person"]
+        filtered_member = filter_member_dict(member)
+        return filtered_member
+    else:
+        return []
+
+
+def get_speeches_data(anftyp, size):
+    """
+    Get speeches data from speech's api
+    and filter for relevant keys.
+    For each speech we query the member api 
+    to get the member data and 
+    then merge it with the speech data.
+    """
+
+    domain = 'http://data.riksdagen.se'
+    speeches = 'anforandelista'
+    format_type = 'json'
+    response = requests.get('{}/{}/?anftyp={}&sz={}&utformat={}'.format(domain, speeches, anftyp, size, format_type))
+    if response.status_code == 200:
+        data = response.json()
+        speeches = data["anforandelista"]["anforande"]
+        filtered_speeches = filter_speeches_dict(speeches)
+        for speech in filtered_speeches:
+            member_data = get_member_data(speech["intressent_id"])        
```

Comment:
What if the `speech["intressent_id"]` is null?

Reply:
updated:

    if response.status_code == 200:
        data = response.json()
        if data["personlista"]["@hits"] == "1":
            member = data["personlista"]["person"]
            filtered_member = filter_member_dict(member)
            return filtered_member
        elif data["personlista"]["@hits"] == "0":
            nullMemberDict = {"member data": "null"}
            return nullMemberDict
    elif response.status_code == 404:
         return not_found
    else:
        return failed